libcontainer/init_linux.go

```diff
@@ -262,11 +314,17 @@ func fixStdioPermissions(u *user.ExecUser) error {
 		if err := syscall.Fstat(int(fd), &s); err != nil {
 			return err
 		}
-		// skip chown of /dev/null if it was used as one of the STDIO fds.
+		// Skip chown of /dev/null if it was used as one of the STDIO fds.
 		if s.Rdev == null.Rdev {
 			continue
 		}
-		if err := syscall.Fchown(int(fd), u.Uid, u.Gid); err != nil {
+		// We only change the uid owner (as it is possible for the mount to
+		// prefer a different gid, and there's no reason for us to change it).
+		// The reason why we don't just leave the default uid=X mount setup is
+		// that users expect to be able to actually use their console. Without
+		// this code, you couldn't effectively run as a non-root user inside a
+		// container and also have a console set up.
+		if err := syscall.Fchown(int(fd), u.Uid, int(s.Gid)); err != nil {
```

Comment:
If `gid=x` is not specified, then the default GID owner is root. Even if you set the GID specifically in `User` what matters is whether you have administrative access to the console. You only need to be the UID owner to do any administrative task, so also requiring the GID owner would be redundant. In addition, the reason this code exists is to allow for things like `wall` and `write` to work -- where you have a special `tty` group that can write to all of the consoles.